libdnf/transaction/Transformer.cpp

```diff
@@ -453,89 +457,101 @@ Transformer::transformRPMItems(SQLite3Ptr swdb,
+        auto len = json_object_array_length(value);
+        for (std::size_t i = 0; i < len; ++i) {
             const char *key = json_object_get_string(json_object_array_get_idx(value, i));
-            compsGroup->addPackage(key, false, CompsPackageType::MANDATORY);
+            auto & pkg = compsGroup->new_package();
+            pkg.set_name(key);
+            pkg.set_installed(false);
+            pkg.set_package_type(CompsPackageType::MANDATORY);
         }
     }
 
-    compsGroup->save();
+    //compsGroup->save();
     return compsGroup;
 }
 
 /**
  * Construct CompsEnvironmentItem object from JSON
  * \param env environment json object
  */
-std::shared_ptr< CompsEnvironmentItem >
-Transformer::processEnvironment(SQLite3Ptr swdb, const char *envId, struct json_object *env)
+std::shared_ptr< CompsEnvironment >
+Transformer::processEnvironment(Transaction & trans, const char *envId, struct json_object *env)
 {
     struct json_object *value;
 
     // create environment
-    auto compsEnv = std::make_shared< CompsEnvironmentItem >(swdb);
-    compsEnv->setEnvironmentId(envId);
+    auto compsEnv = std::make_shared< CompsEnvironment >(trans);
+    compsEnv->set_environment_id(envId);
 
     if (json_object_object_get_ex (env, "name", &value)) {
-        compsEnv->setName(json_object_get_string(value));
+        compsEnv->set_name(json_object_get_string(value));
     }
 
     if (json_object_object_get_ex (env, "ui_name", &value)) {
-        compsEnv->setTranslatedName(json_object_get_string(value));
+        compsEnv->set_translated_name(json_object_get_string(value));
     }
 
     // TODO parse pkg_types/grp_types to CompsPackageType
     if (json_object_object_get_ex(env, "full_list", &value)) {
-        int len = json_object_array_length(value);
-        for (int i = 0; i < len; ++i) {
+        auto len = json_object_array_length(value);
+        for (std::size_t i = 0; i < len; ++i) {
             const char *key = json_object_get_string(json_object_array_get_idx(value, i));
-            compsEnv->addGroup(key, true, CompsPackageType::MANDATORY);
+            auto & grp = compsEnv->new_group();
+            grp.set_group_id(key);
+            grp.set_installed(true);
+            grp.set_group_type(CompsPackageType::MANDATORY);
         }
     }
 
     // TODO parse pkg_types/grp_types to CompsPackageType
     if (json_object_object_get_ex(env, "pkg_exclude", &value)) {
-        int len = json_object_array_length(value);
-        for (int i = 0; i < len; ++i) {
+        auto len = json_object_array_length(value);
+        for (std::size_t i = 0; i < len; ++i) {
             const char *key = json_object_get_string(json_object_array_get_idx(value, i));
-            compsEnv->addGroup(key, false, CompsPackageType::MANDATORY);
+            auto & grp = compsEnv->new_group();
+            grp.set_group_id(key);
+            grp.set_installed(false);
+            grp.set_group_type(CompsPackageType::MANDATORY);
```

Comment:
Let me think. In case that constructor of EnvGroup will be private and also method compsEnv->new_group(), then we can keep the original interface `compsEnv->add_group(key, false, CompsPackageType::MANDATORY);`. A also suggest to rename `compsEnv` to `comps_env`.